[CI] Fix Minari tests

younik opened the pull request:
## Description

With Minari 0.5.0 release, we regrouped dependencies to install only the required ones depending on user usage. 
Moreover, we restructured the datasets to be hierarchical. This caused the `torch/rl` tests to fail; this PR should fix them.

I changed the tests to rely on Minari for getting the last version of the datasets.

Also, Minari 0.5.0 adds support for Arrow datasets, I am happy to integrate it in `torch/rl`.

## Motivation and Context

Fixes failing CI due to new version of Minari.

- [ ] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

vmoens commented:
We now have multiple. `FAILED test/test_libs.py::TestMinari::test_load[D4RL/door/human-v2-True] - ModuleNotFoundError: No module named 'gymnasium_robotics'` in the CI.

The linter (pre-commit) is also unhappy.

https://github.com/pytorch/rl/blob/0326c412274f6304e59d6cd8c9447d3b54a06213/CONTRIBUTING.md?plain=1#L34-L52

younik commented:
> We now have multiple. `FAILED test/test_libs.py::TestMinari::test_load[D4RL/door/human-v2-True] - ModuleNotFoundError: No module named 'gymnasium_robotics'` in the CI.
> 
> The linter (pre-commit) is also unhappy.
> 
> https://github.com/pytorch/rl/blob/0326c412274f6304e59d6cd8c9447d3b54a06213/CONTRIBUTING.md?plain=1#L34-L52

Ups, my env was contaminated. 
The dependency error is due to [a line](https://github.com/Farama-Foundation/Minari/blob/be4ee5b5471d7152c3eea53650e9b7eb13752112/minari/storage/hosting.py#L174) incorrectly unreachable in `0.4.3`. However, throwing a dependency error during download is not the intended behavior, and I will fix it in Minari. I will release it this week.

younik commented:
After further analysis, the issue was with remote datasets lacking spaces, causing Minari to fall back to `env_spec`. This has been fixed. Could you please re-run the Minari tests in CI? @vmoens

younik commented on test/test_libs.py:
```diff
@@ -2823,34 +2823,11 @@ def _minari_selected_datasets():
-    # We rely on sorting the keys as v0 < v1 but if the version is greater than 9 this won't work
-    total_keys = sorted(minari.list_remote_datasets())
-    assert not any(
-        key[-2:] == "10" for key in total_keys
-    ), "You should adapt the Minari test scripts as some dataset have a version >= 10 and sorting will fail."
-    total_keys_splits = [key.split("-") for key in total_keys]
+    total_keys = sorted(
+        minari.list_remote_datasets(latest_version=True, compatible_minari_version=True)
+    )
     indices = torch.randperm(len(total_keys))[:20]
     keys = [total_keys[idx] for idx in indices]
-    keys = [
-        key
-        for key in keys
-        if "=0.4" in minari.list_remote_datasets()[key]["minari_version"]
-    ]
-
-    def _replace_with_max(key):
-        key_split = key.split("-")
-        same_entries = (
-            torch.tensor(
-                [total_key[:-1] == key_split[:-1] for total_key in total_keys_splits]
-            )
-            .nonzero()
-            .squeeze()
-            .tolist()
-        )
-        last_same_entry = same_entries[-1]
-        return total_keys[last_same_entry]
-
-    keys = [_replace_with_max(key) for key in keys]
```
I believed what changed is the order of _MINARI_DATASETS given the above modification.
The test work with `kitchen` datasets as the observation space is a dictionary, while it doesn't work for `pen` datasets.
I can change the code to test only a kitchen dataset or to work with the pen one.

younik commented:
I changed the preprocessing test to only use a dataset with Dict observation space.
I have also updated the kitchen datasets that have wrong info structure.
